spec/sales_engine_spec.rb

```diff
@@ -1,22 +1,43 @@
 require 'CSV'
 require 'rspec'
 
+require './data/invoice_mocks'
 require './data/item_mocks'
 require './data/merchant_mocks'
+require './data/Mockable'
 require './lib/file_io'
+require './lib/invoice_repository'
 require './lib/item_repository'
 require './lib/merchant_repository'
 require './lib/sales_engine'
 require './lib/sales_analyst'
+require './lib/transaction_repository'
 
 describe SalesEngine do
+  details = {
+    id: 6,
+    invoice_id: 8,
+    credit_card_number: '4242424242424242',
+    credit_card_expiration_date: '0220',
+    result: 'success',
+    created_at: Time.now,
+    updated_at: Time.now
+  }
+  data_hashes = []
+  10.times { data_hashes << details }
+
   describe '#from_csv' do
     it 'creates a new instance of SalesEngine' do
       mock_items = ItemMocks.items_as_mocks(self)
       allow_any_instance_of(ItemRepository).to receive(:create_items).and_return(mock_items)
       mock_merchants = MerchantMocks.merchants_as_mocks(self)
       allow_any_instance_of(MerchantRepository).to receive(:create_merchants).and_return(mock_merchants)
-      files = { items: './file1.csv', merchants: './file2.csv' }
```

Comment:
Are we going to run into merge conflicts between this and pull #88? They wouldn't be hard to resolve but we're definitely touching the same file here

Reply:
I just merged this branch into that one @NoahZinter so as long as we merge this one first, it'll be perfectly fine. I'll resolve any issues in there before we merge that in (don't think there should be any though).